### PR TITLE
Introduce bin/upgrade to auto-upgrade libraries

### PR DESCRIPTION
**Pull request by mbryzek**
- fetches recommendations from dependency app and auto
   applies to local build.sbt file

### CODE REVIEW

**Comment by paololim** [comment]
Does the README.md need to change?

File: bin/upgrade [hunk 1]
```diff
@@ -19,66 +76,41 @@ if !File.exists?(FILE)
...
-    commit_command = "git commit -m 'Upgrade %s.%s from %s => %s' %s" % [group_id, artifact_id, old_version, version, FILE]
-    if commit
-      [commit_command, "git push origin master"].each do |cmd|
-        puts cmd
-        system(cmd)
-      end
     else
-      puts "Sample commit command:"
-      puts commit_command
+      puts " - Skipping recommendation for %s (unknown type)" % rec.name
```
Comment (paololim) [inline bin/upgrade]: What are the unknown types? Why would it be unknown?

File: bin/upgrade-library [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    exit(1)
+  elsif old_version == version
+    puts "#{FILE} already references version #{version}"
+  else
+    puts "Rewriting #{FILE} with updated version #{version}"
+    File.rename(tmp, FILE)
+
+    commit_command = "git commit -m 'Upgrade %s.%s from %s => %s' %s" % [group_id, artifact_id, old_version, version, FILE]
+    if commit
+      [commit_command, "git push origin master"].each do |cmd|
```
Comment (paololim) [inline bin/upgrade-library]: This seems to assume we are on the master branch of the repo. Perhaps we could add a `git checkout master` somewhere earlier just to make sure this is the case.

File: lib/tools/config.rb [hunk 3]
```diff
@@ -142,6 +157,10 @@ def docker_password
       @data[:docker_password]
     end
 
+    def dependency_api_token
```
Comment (paololim) [inline lib/tools/config.rb]: Might be good to update the readme with how to set this up

File: bin/upgrade (hunk 1 above)
Comment (mbryzek) [inline bin/upgrade]: none right now. this is for forward compatibility - apidoc always includes an undefined type - so I rather print the warning then do nothing

File: bin/upgrade-library (hunk 2 above)
Comment (mbryzek) [inline bin/upgrade-library]: agreed - going to assert we are on master before pushing

File: lib/tools/config.rb (hunk 3 above)
Comment (mbryzek) [inline lib/tools/config.rb]: done

**Comment by paololim** [comment]
Lgtm

**Pull request merged**